instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteAlternativesTest.kt

```diff
@@ -107,7 +119,58 @@ class RouteAlternativesTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::cla
         val countDownLatch = CountDownLatch(1)
         runOnMainSync {
             mapboxNavigation.historyRecorder.stopRecording {
-                logE("history path=$it", "DEBUG")
+                logE("history path=$it", LOG_CATEGORY)
+                countDownLatch.countDown()
+            }
+        }
+        countDownLatch.await()
+
+        // Verify alternative routes events were triggered.
+        assertEquals(2, routes.size)
+        assertTrue(mapboxNavigation.getNavigationRoutes().isNotEmpty())
+        firstAlternative.verifyOnRouteAlternativesAndProgressReceived()
+        firstAlternative.alternatives!!.forEach {
+            assertFalse(routes.contains(it))
+        }
+    }
+
+    @Test
+    fun alternative_routes_observer_is_not_triggered_for_set_routes() {
+        // Prepare with alternative routes.
+        setupMockRequestHandlers(coordinates)
+        val routes = requestDirectionsRouteSync(coordinates)
+
+        // Play primary route
+        runOnMainSync {
+            mapboxNavigation.historyRecorder.startRecording()
+            mockLocationReplayerRule.playRoute(routes.first())
+            mapboxNavigation.startTripSession()
+        }
+
+        // Wait for enhanced locations to start and then set the routes.
+        val firstLocationIdlingResource = FirstLocationIdlingResource(mapboxNavigation)
+        firstLocationIdlingResource.firstLocationSync()
+        runOnMainSync {
+            mapboxNavigation.setRoutes(routes)
+        }
+
+        // Subscribing for alternatives
+        val firstAlternative = RouteAlternativesIdlingResource(
+            mapboxNavigation
+        ) { _, alternatives, _ ->
+            alternatives.isNotEmpty()
+        }
+        firstAlternative.register()
```

Comment:
This should be done before
```
mapboxNavigation.setRoutes(routes)
```

Otherwise, if the alternatives observer was caught synchronously, we wouldn't catch it.

Reply:
make sense